python/ray/data/_internal/planner/plan_write_op.py

```diff
@@ -16,19 +18,38 @@
 from ray.data.datasource.datasource import Datasource
 
 
+def gen_data_sink_write_result(
```

Comment:
Nit: `Datasink` is one word, so this seems more accurate?
```suggestion
def gen_datasink_write_result(
```